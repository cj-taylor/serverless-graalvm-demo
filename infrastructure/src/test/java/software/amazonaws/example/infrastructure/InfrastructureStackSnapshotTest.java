package software.amazonaws.example.infrastructure;

import java.io.IOException;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import software.amazon.awscdk.App;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.assertions.Template;

@ExtendWith(SnapshotExtension.class)
public class InfrastructureStackSnapshotTest {

    private Expect expect;

    private static Template template;

    @BeforeAll
    static void setup() {
        var app = new App();

        var stack = new InfrastructureStack(app, "test", StackProps.builder().build());

        template = Template.fromStack(stack);
    }

    @Test
    public void testSnapshot() throws IOException {
        expect.toMatchSnapshot(template.toJSON());
    }
}
