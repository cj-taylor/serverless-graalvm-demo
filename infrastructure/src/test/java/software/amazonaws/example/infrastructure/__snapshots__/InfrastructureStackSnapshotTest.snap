software.amazonaws.example.infrastructure.InfrastructureStackSnapshotTest.testSnapshot=[
  {
    "Outputs": {
      "ApiUrl": {
        "Export": {
          "Name": "ApiUrl"
        },
        "Value": {
          "Fn::GetAtt": [
            "ProductsApiEC66475F",
            "ApiEndpoint"
          ]
        }
      }
    },
    "Parameters": {
      "BootstrapVersion": {
        "Default": "/cdk-bootstrap/hnb659fds/version",
        "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
        "Type": "AWS::SSM::Parameter::Value<String>"
      }
    },
    "Resources": {
      "DeleteProductFunctionDCB5D2F5": {
        "DependsOn": [
          "DeleteProductFunctionServiceRoleDefaultPolicy3A4E8B57",
          "DeleteProductFunctionServiceRole522EE7CE"
        ],
        "Properties": {
          "Code": {
            "S3Bucket": {
              "Fn::Sub": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}"
            },
            "S3Key": "6f93ef7f4500f963dcb066daf0b796c7c094fc1eec4aa647b0d7344cf43c9683.zip"
          },
          "Environment": {
            "Variables": {
              "PRODUCT_TABLE_NAME": {
                "Ref": "Products229621C6"
              }
            }
          },
          "Handler": "software.amazonaws.example.product.entrypoints.ApiGatewayDeleteProductRequestHandler",
          "MemorySize": 256,
          "Role": {
            "Fn::GetAtt": [
              "DeleteProductFunctionServiceRole522EE7CE",
              "Arn"
            ]
          },
          "Runtime": "provided.al2",
          "TracingConfig": {
            "Mode": "Active"
          }
        },
        "Type": "AWS::Lambda::Function"
      },
      "DeleteProductFunctionLogRetentionFF8537DE": {
        "Properties": {
          "LogGroupName": {
            "Fn::Join": [
              "",
              [
                "/aws/lambda/",
                {
                  "Ref": "DeleteProductFunctionDCB5D2F5"
                }
              ]
            ]
          },
          "RetentionInDays": 7,
          "ServiceToken": {
            "Fn::GetAtt": [
              "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
              "Arn"
            ]
          }
        },
        "Type": "Custom::LogRetention"
      },
      "DeleteProductFunctionServiceRole522EE7CE": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "lambda.amazonaws.com"
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                ]
              ]
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "DeleteProductFunctionServiceRoleDefaultPolicy3A4E8B57": {
        "Properties": {
          "PolicyDocument": {
            "Statement": [
              {
                "Action": [
                  "xray:PutTraceSegments",
                  "xray:PutTelemetryRecords"
                ],
                "Effect": "Allow",
                "Resource": "*"
              },
              {
                "Action": [
                  "dynamodb:BatchWriteItem",
                  "dynamodb:PutItem",
                  "dynamodb:UpdateItem",
                  "dynamodb:DeleteItem"
                ],
                "Effect": "Allow",
                "Resource": [
                  {
                    "Fn::GetAtt": [
                      "Products229621C6",
                      "Arn"
                    ]
                  },
                  {
                    "Ref": "AWS::NoValue"
                  }
                ]
              }
            ],
            "Version": "2012-10-17"
          },
          "PolicyName": "DeleteProductFunctionServiceRoleDefaultPolicy3A4E8B57",
          "Roles": [
            {
              "Ref": "DeleteProductFunctionServiceRole522EE7CE"
            }
          ]
        },
        "Type": "AWS::IAM::Policy"
      },
      "GetAllProductFunction9216DFAB": {
        "DependsOn": [
          "GetAllProductFunctionServiceRoleDefaultPolicy1D99BC2B",
          "GetAllProductFunctionServiceRoleB839C02A"
        ],
        "Properties": {
          "Code": {
            "S3Bucket": {
              "Fn::Sub": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}"
            },
            "S3Key": "6f93ef7f4500f963dcb066daf0b796c7c094fc1eec4aa647b0d7344cf43c9683.zip"
          },
          "Environment": {
            "Variables": {
              "PRODUCT_TABLE_NAME": {
                "Ref": "Products229621C6"
              }
            }
          },
          "Handler": "software.amazonaws.example.product.entrypoints.ApiGatewayGetAllProductRequestHandler",
          "MemorySize": 256,
          "Role": {
            "Fn::GetAtt": [
              "GetAllProductFunctionServiceRoleB839C02A",
              "Arn"
            ]
          },
          "Runtime": "provided.al2",
          "TracingConfig": {
            "Mode": "Active"
          }
        },
        "Type": "AWS::Lambda::Function"
      },
      "GetAllProductFunctionLogRetention9ED8A84E": {
        "Properties": {
          "LogGroupName": {
            "Fn::Join": [
              "",
              [
                "/aws/lambda/",
                {
                  "Ref": "GetAllProductFunction9216DFAB"
                }
              ]
            ]
          },
          "RetentionInDays": 7,
          "ServiceToken": {
            "Fn::GetAtt": [
              "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
              "Arn"
            ]
          }
        },
        "Type": "Custom::LogRetention"
      },
      "GetAllProductFunctionServiceRoleB839C02A": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "lambda.amazonaws.com"
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                ]
              ]
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "GetAllProductFunctionServiceRoleDefaultPolicy1D99BC2B": {
        "Properties": {
          "PolicyDocument": {
            "Statement": [
              {
                "Action": [
                  "xray:PutTraceSegments",
                  "xray:PutTelemetryRecords"
                ],
                "Effect": "Allow",
                "Resource": "*"
              },
              {
                "Action": [
                  "dynamodb:BatchGetItem",
                  "dynamodb:GetRecords",
                  "dynamodb:GetShardIterator",
                  "dynamodb:Query",
                  "dynamodb:GetItem",
                  "dynamodb:Scan",
                  "dynamodb:ConditionCheckItem"
                ],
                "Effect": "Allow",
                "Resource": [
                  {
                    "Fn::GetAtt": [
                      "Products229621C6",
                      "Arn"
                    ]
                  },
                  {
                    "Ref": "AWS::NoValue"
                  }
                ]
              }
            ],
            "Version": "2012-10-17"
          },
          "PolicyName": "GetAllProductFunctionServiceRoleDefaultPolicy1D99BC2B",
          "Roles": [
            {
              "Ref": "GetAllProductFunctionServiceRoleB839C02A"
            }
          ]
        },
        "Type": "AWS::IAM::Policy"
      },
      "GetProductFunction879D9990": {
        "DependsOn": [
          "GetProductFunctionServiceRoleDefaultPolicy128C2561",
          "GetProductFunctionServiceRole29A2DC57"
        ],
        "Properties": {
          "Code": {
            "S3Bucket": {
              "Fn::Sub": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}"
            },
            "S3Key": "6f93ef7f4500f963dcb066daf0b796c7c094fc1eec4aa647b0d7344cf43c9683.zip"
          },
          "Environment": {
            "Variables": {
              "PRODUCT_TABLE_NAME": {
                "Ref": "Products229621C6"
              }
            }
          },
          "Handler": "software.amazonaws.example.product.entrypoints.ApiGatewayGetProductRequestHandler",
          "MemorySize": 256,
          "Role": {
            "Fn::GetAtt": [
              "GetProductFunctionServiceRole29A2DC57",
              "Arn"
            ]
          },
          "Runtime": "provided.al2",
          "TracingConfig": {
            "Mode": "Active"
          }
        },
        "Type": "AWS::Lambda::Function"
      },
      "GetProductFunctionLogRetention07349A0C": {
        "Properties": {
          "LogGroupName": {
            "Fn::Join": [
              "",
              [
                "/aws/lambda/",
                {
                  "Ref": "GetProductFunction879D9990"
                }
              ]
            ]
          },
          "RetentionInDays": 7,
          "ServiceToken": {
            "Fn::GetAtt": [
              "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
              "Arn"
            ]
          }
        },
        "Type": "Custom::LogRetention"
      },
      "GetProductFunctionServiceRole29A2DC57": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "lambda.amazonaws.com"
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                ]
              ]
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "GetProductFunctionServiceRoleDefaultPolicy128C2561": {
        "Properties": {
          "PolicyDocument": {
            "Statement": [
              {
                "Action": [
                  "xray:PutTraceSegments",
                  "xray:PutTelemetryRecords"
                ],
                "Effect": "Allow",
                "Resource": "*"
              },
              {
                "Action": [
                  "dynamodb:BatchGetItem",
                  "dynamodb:GetRecords",
                  "dynamodb:GetShardIterator",
                  "dynamodb:Query",
                  "dynamodb:GetItem",
                  "dynamodb:Scan",
                  "dynamodb:ConditionCheckItem"
                ],
                "Effect": "Allow",
                "Resource": [
                  {
                    "Fn::GetAtt": [
                      "Products229621C6",
                      "Arn"
                    ]
                  },
                  {
                    "Ref": "AWS::NoValue"
                  }
                ]
              }
            ],
            "Version": "2012-10-17"
          },
          "PolicyName": "GetProductFunctionServiceRoleDefaultPolicy128C2561",
          "Roles": [
            {
              "Ref": "GetProductFunctionServiceRole29A2DC57"
            }
          ]
        },
        "Type": "AWS::IAM::Policy"
      },
      "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A": {
        "DependsOn": [
          "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
          "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB"
        ],
        "Properties": {
          "Code": {
            "S3Bucket": {
              "Fn::Sub": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}"
            },
            "S3Key": "b120b13d9d868c7622e7db1b68bae4c0f82ffd0227b8c15f2cef38e186ff3827.zip"
          },
          "Handler": "index.handler",
          "Role": {
            "Fn::GetAtt": [
              "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB",
              "Arn"
            ]
          },
          "Runtime": "nodejs14.x"
        },
        "Type": "AWS::Lambda::Function"
      },
      "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "lambda.amazonaws.com"
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                ]
              ]
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB": {
        "Properties": {
          "PolicyDocument": {
            "Statement": [
              {
                "Action": [
                  "logs:PutRetentionPolicy",
                  "logs:DeleteRetentionPolicy"
                ],
                "Effect": "Allow",
                "Resource": "*"
              }
            ],
            "Version": "2012-10-17"
          },
          "PolicyName": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRoleDefaultPolicyADDA7DEB",
          "Roles": [
            {
              "Ref": "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aServiceRole9741ECFB"
            }
          ]
        },
        "Type": "AWS::IAM::Policy"
      },
      "Products229621C6": {
        "DeletionPolicy": "Retain",
        "Properties": {
          "AttributeDefinitions": [
            {
              "AttributeName": "PK",
              "AttributeType": "S"
            }
          ],
          "BillingMode": "PAY_PER_REQUEST",
          "KeySchema": [
            {
              "AttributeName": "PK",
              "KeyType": "HASH"
            }
          ],
          "TableName": "Products"
        },
        "Type": "AWS::DynamoDB::Table",
        "UpdateReplacePolicy": "Retain"
      },
      "ProductsApiDELETEidA42E3445": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "AuthorizationType": "NONE",
          "RouteKey": "DELETE /{id}",
          "Target": {
            "Fn::Join": [
              "",
              [
                "integrations/",
                {
                  "Ref": "ProductsApiDELETEidHttpIntegration5a548d709ada26f8f49e90a3bf96395a6119D68E"
                }
              ]
            ]
          }
        },
        "Type": "AWS::ApiGatewayV2::Route"
      },
      "ProductsApiDELETEidHttpIntegration5a548d709ada26f8f49e90a3bf96395a6119D68E": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "IntegrationType": "AWS_PROXY",
          "IntegrationUri": {
            "Fn::GetAtt": [
              "DeleteProductFunctionDCB5D2F5",
              "Arn"
            ]
          },
          "PayloadFormatVersion": "2.0"
        },
        "Type": "AWS::ApiGatewayV2::Integration"
      },
      "ProductsApiDELETEidtestProductsApiDELETEidD6B80B65PermissionEC2856A6": {
        "Properties": {
          "Action": "lambda:InvokeFunction",
          "FunctionName": {
            "Fn::GetAtt": [
              "DeleteProductFunctionDCB5D2F5",
              "Arn"
            ]
          },
          "Principal": "apigateway.amazonaws.com",
          "SourceArn": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":execute-api:",
                {
                  "Ref": "AWS::Region"
                },
                ":",
                {
                  "Ref": "AWS::AccountId"
                },
                ":",
                {
                  "Ref": "ProductsApiEC66475F"
                },
                "/*/*/{id}"
              ]
            ]
          }
        },
        "Type": "AWS::Lambda::Permission"
      },
      "ProductsApiDefaultStageCA0B4BDD": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "AutoDeploy": true,
          "StageName": "$default"
        },
        "Type": "AWS::ApiGatewayV2::Stage"
      },
      "ProductsApiEC66475F": {
        "Properties": {
          "Name": "ProductsApi",
          "ProtocolType": "HTTP"
        },
        "Type": "AWS::ApiGatewayV2::Api"
      },
      "ProductsApiGET64FCA6F5": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "AuthorizationType": "NONE",
          "RouteKey": "GET /",
          "Target": {
            "Fn::Join": [
              "",
              [
                "integrations/",
                {
                  "Ref": "ProductsApiGETHttpIntegrationf79666a9d705df69fb16bec52c7af2bc9AFCE66F"
                }
              ]
            ]
          }
        },
        "Type": "AWS::ApiGatewayV2::Route"
      },
      "ProductsApiGETHttpIntegrationf79666a9d705df69fb16bec52c7af2bc9AFCE66F": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "IntegrationType": "AWS_PROXY",
          "IntegrationUri": {
            "Fn::GetAtt": [
              "GetAllProductFunction9216DFAB",
              "Arn"
            ]
          },
          "PayloadFormatVersion": "2.0"
        },
        "Type": "AWS::ApiGatewayV2::Integration"
      },
      "ProductsApiGETid0983AC5E": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "AuthorizationType": "NONE",
          "RouteKey": "GET /{id}",
          "Target": {
            "Fn::Join": [
              "",
              [
                "integrations/",
                {
                  "Ref": "ProductsApiGETidHttpIntegration38b48bd2cf87fb66995db50b012df11d1293D6B2"
                }
              ]
            ]
          }
        },
        "Type": "AWS::ApiGatewayV2::Route"
      },
      "ProductsApiGETidHttpIntegration38b48bd2cf87fb66995db50b012df11d1293D6B2": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "IntegrationType": "AWS_PROXY",
          "IntegrationUri": {
            "Fn::GetAtt": [
              "GetProductFunction879D9990",
              "Arn"
            ]
          },
          "PayloadFormatVersion": "2.0"
        },
        "Type": "AWS::ApiGatewayV2::Integration"
      },
      "ProductsApiGETidtestProductsApiGETidAF51EDF6Permission02108EA6": {
        "Properties": {
          "Action": "lambda:InvokeFunction",
          "FunctionName": {
            "Fn::GetAtt": [
              "GetProductFunction879D9990",
              "Arn"
            ]
          },
          "Principal": "apigateway.amazonaws.com",
          "SourceArn": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":execute-api:",
                {
                  "Ref": "AWS::Region"
                },
                ":",
                {
                  "Ref": "AWS::AccountId"
                },
                ":",
                {
                  "Ref": "ProductsApiEC66475F"
                },
                "/*/*/{id}"
              ]
            ]
          }
        },
        "Type": "AWS::Lambda::Permission"
      },
      "ProductsApiGETtestProductsApiGETA29D8F69Permission1AAAC8A7": {
        "Properties": {
          "Action": "lambda:InvokeFunction",
          "FunctionName": {
            "Fn::GetAtt": [
              "GetAllProductFunction9216DFAB",
              "Arn"
            ]
          },
          "Principal": "apigateway.amazonaws.com",
          "SourceArn": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":execute-api:",
                {
                  "Ref": "AWS::Region"
                },
                ":",
                {
                  "Ref": "AWS::AccountId"
                },
                ":",
                {
                  "Ref": "ProductsApiEC66475F"
                },
                "/*/*/"
              ]
            ]
          }
        },
        "Type": "AWS::Lambda::Permission"
      },
      "ProductsApiPUTid87C3D32E": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "AuthorizationType": "NONE",
          "RouteKey": "PUT /{id}",
          "Target": {
            "Fn::Join": [
              "",
              [
                "integrations/",
                {
                  "Ref": "ProductsApiPUTidHttpIntegration6e96aaaacc89d4a5d4f1fc6205164ae6C14DD3C4"
                }
              ]
            ]
          }
        },
        "Type": "AWS::ApiGatewayV2::Route"
      },
      "ProductsApiPUTidHttpIntegration6e96aaaacc89d4a5d4f1fc6205164ae6C14DD3C4": {
        "Properties": {
          "ApiId": {
            "Ref": "ProductsApiEC66475F"
          },
          "IntegrationType": "AWS_PROXY",
          "IntegrationUri": {
            "Fn::GetAtt": [
              "PutProductFunctionC4AE1EE6",
              "Arn"
            ]
          },
          "PayloadFormatVersion": "2.0"
        },
        "Type": "AWS::ApiGatewayV2::Integration"
      },
      "ProductsApiPUTidtestProductsApiPUTidF5B3415BPermissionEC0137DA": {
        "Properties": {
          "Action": "lambda:InvokeFunction",
          "FunctionName": {
            "Fn::GetAtt": [
              "PutProductFunctionC4AE1EE6",
              "Arn"
            ]
          },
          "Principal": "apigateway.amazonaws.com",
          "SourceArn": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":execute-api:",
                {
                  "Ref": "AWS::Region"
                },
                ":",
                {
                  "Ref": "AWS::AccountId"
                },
                ":",
                {
                  "Ref": "ProductsApiEC66475F"
                },
                "/*/*/{id}"
              ]
            ]
          }
        },
        "Type": "AWS::Lambda::Permission"
      },
      "PutProductFunctionC4AE1EE6": {
        "DependsOn": [
          "PutProductFunctionServiceRoleDefaultPolicyA7672A29",
          "PutProductFunctionServiceRole213C5CA6"
        ],
        "Properties": {
          "Code": {
            "S3Bucket": {
              "Fn::Sub": "cdk-hnb659fds-assets-${AWS::AccountId}-${AWS::Region}"
            },
            "S3Key": "6f93ef7f4500f963dcb066daf0b796c7c094fc1eec4aa647b0d7344cf43c9683.zip"
          },
          "Environment": {
            "Variables": {
              "PRODUCT_TABLE_NAME": {
                "Ref": "Products229621C6"
              }
            }
          },
          "Handler": "software.amazonaws.example.product.entrypoints.ApiGatewayPutProductRequestHandler",
          "MemorySize": 256,
          "Role": {
            "Fn::GetAtt": [
              "PutProductFunctionServiceRole213C5CA6",
              "Arn"
            ]
          },
          "Runtime": "provided.al2",
          "TracingConfig": {
            "Mode": "Active"
          }
        },
        "Type": "AWS::Lambda::Function"
      },
      "PutProductFunctionLogRetention11C3A212": {
        "Properties": {
          "LogGroupName": {
            "Fn::Join": [
              "",
              [
                "/aws/lambda/",
                {
                  "Ref": "PutProductFunctionC4AE1EE6"
                }
              ]
            ]
          },
          "RetentionInDays": 7,
          "ServiceToken": {
            "Fn::GetAtt": [
              "LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8aFD4BFC8A",
              "Arn"
            ]
          }
        },
        "Type": "Custom::LogRetention"
      },
      "PutProductFunctionServiceRole213C5CA6": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "lambda.amazonaws.com"
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                ]
              ]
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      },
      "PutProductFunctionServiceRoleDefaultPolicyA7672A29": {
        "Properties": {
          "PolicyDocument": {
            "Statement": [
              {
                "Action": [
                  "xray:PutTraceSegments",
                  "xray:PutTelemetryRecords"
                ],
                "Effect": "Allow",
                "Resource": "*"
              },
              {
                "Action": [
                  "dynamodb:BatchWriteItem",
                  "dynamodb:PutItem",
                  "dynamodb:UpdateItem",
                  "dynamodb:DeleteItem"
                ],
                "Effect": "Allow",
                "Resource": [
                  {
                    "Fn::GetAtt": [
                      "Products229621C6",
                      "Arn"
                    ]
                  },
                  {
                    "Ref": "AWS::NoValue"
                  }
                ]
              }
            ],
            "Version": "2012-10-17"
          },
          "PolicyName": "PutProductFunctionServiceRoleDefaultPolicyA7672A29",
          "Roles": [
            {
              "Ref": "PutProductFunctionServiceRole213C5CA6"
            }
          ]
        },
        "Type": "AWS::IAM::Policy"
      }
    },
    "Rules": {
      "CheckBootstrapVersion": {
        "Assertions": [
          {
            "Assert": {
              "Fn::Not": [
                {
                  "Fn::Contains": [
                    [
                      "1",
                      "2",
                      "3",
                      "4",
                      "5"
                    ],
                    {
                      "Ref": "BootstrapVersion"
                    }
                  ]
                }
              ]
            },
            "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI."
          }
        ]
      }
    }
  }
]